yt_dlp/extractor/subsplash.py

```diff
@@ -0,0 +1,186 @@
+import functools
+import json
+import math
+
+from .common import InfoExtractor
+from ..utils import (
+    InAdvancePagedList,
+    get_element_by_id,
+    int_or_none,
+    parse_iso8601,
+    try_call,
+    url_or_none,
+)
+from ..utils.traversal import traverse_obj
+
+
+class SubsplashBaseIE(InfoExtractor):
+    def _get_headers(self, url, display_id):
+        token = try_call(lambda: self._get_cookies(url)['ss-token-guest'].value)
+        if not token:
+            webpage, urlh = self._download_webpage_handle(url, display_id)
+            token = try_call(lambda: self._get_cookies(url)['ss-token-guest'].value)
+            if not token:
+                token = urlh.get_header('x-api-token')
+            if not token:
+                token = traverse_obj(get_element_by_id('shoebox-tokens', webpage), ({json.loads}, 'apiToken', {str}))
+            if not token:
+                token = self._search_regex(r'\\"tokens\\":{\\"guest\\":\\"([A-Za-z0-9._-]+)\\"', webpage, 'token', default=None)
+
+        if not token:
+            return None
+        return {'Authorization': f'Bearer {token}'}
+
+    def _extract_video(self, data, display_id):
+        formats = []
+        video_data = traverse_obj(data, ('_embedded', 'video', '_embedded', {dict}))
+        m3u8_url = traverse_obj(video_data, ('playlists', 0, '_links', 'related', 'href', {url_or_none}))
+        if m3u8_url:
+            formats.extend(self._extract_m3u8_formats(m3u8_url, display_id, 'mp4', m3u8_id='hls', fatal=False))
+        mp4_entry = traverse_obj(video_data, ('video-outputs', lambda _, v: url_or_none(v['_links']['related']['href']), any))
+        if mp4_entry:
+            formats.append({
+                'url': mp4_entry['_links']['related']['href'],
+                'format_id': 'direct',
+                'preference': 1,
```

Comment:
`preference` can't be overridden by the user's format sorting. So we'd only want to set this if the direct format URL is always the highest resolution/etc available

Reply:
It is, but I wasn't aware that it overrides everything. In that case, it should imo. never be used. Using `preference` even breaks `-S +res `